marklogic-data-hub-central/ui/src/App.tsx

```diff
@@ -53,9 +63,8 @@ const App: React.FC<Props> = ({history, location}) => {
     } else {
       if (user.error.type !== '') {
         history.push('/error');
-      } else if (location.pathname !== '/' && location.pathname !== '/noresponse') {
-        user.pageRoute = location.pathname;
       }
+      user.pageRoute = getPageRoute(location);
```

Comment:
If bookmark url contains `/tiles/explore/detail`, authentication leads to 500 Internal Server Error, since docUri seems to be undefined: `http://localhost:3000/api/entitySearch?docUri=undefined`

Detail component gets the `uri` from either location.state or from user preferences in LocalStorage. If the LocalStorage for the user doesnt include `uri`, we will get 500 Internal Server Error. Pinging @rahulvudutala  and @Sanjeevani19 for awarerness.

Reply:
Recently @xnikhil08 made changes to the uri as a part of this PR https://github.com/marklogic/marklogic-data-hub/pull/4489, he might have a better idea about this.

Reply:
@bsrikan - are you still getting this issue?

Reply:
@xnikhil08 yes, I am able to reproduce the 500 error when trying to login as a user who saved the bookmark, when there is no local storage. Its also easily reproducible when the bookmark is used by a different user.

Reply:
Can we fix this by either 1. moving necessary information to the URL or 2. redirecting to a different page if the page if docUri is undefined?

Reply:
@bsrikan I addressed this error by redirecting to Explore when there is no URI accessible for the Detail view.

Reply:
Thanks @wooldridge . Fixed login e2e test to reflect the change.